### DISCUSSION TITLE
How to use SSR, SSG, CSR, ISR methods interchangeably in a Multitenant App for a Specific Page.

### DISCUSSION THREAD

**Discussion by RiyasaatAhmed** [vercel/community#963] [Help] [answered]
Hello there,

We have a Multitenant Ecommerce app utilizing NEXT version 9 and can only support up to 2k items (pages w/45 min-Vercel build limitation—via ISR). Therefore, and due to this limitation, we are updating to 12. In doing so, our goal is to support 100+ clients through a single code base.

We know we'll have different clients, each with a different number of products ranging from 50 to 400k SKUs.

In scenario #1, we have projects with ~200k pages. And 50k or so pages automatically update at a minimum of a 24-hour interval and as often as a 15-minute interval.

In scenario #2, we have projects with ~400 pages where updates happen sporadically. An admin user updates the products manually.

We want to use the SSR, SSG, CSR, ISR, and OnDemandISR methods interchangeably via an environment variable to maintain a single code base. How can we achieve this?

What are the most suitable page rendering techniques for the above scenarios (considering API call optimization)?

In advance, we appreciate your assistance.

**Comment by amyegan** [comment, marked as answer]
Thanks for reaching out @RiyasaatAhmed! This community is generally more focused on questions and issues related to the Vercel platform. In general, I would recommend reaching out to the Next.js community on [Discord](http://nextjs.org/discord) or in [Next.js GitHub Discussions](https://github.com/vercel/next.js/discussions) for help with Next.js questions. 

If you need more personalized assistance with plan limits and solutions, you can [talk to a Vercel expert](https://vercel.com/contact/sales) who will help you find the right plan and pricing for your business.

**Reply by RiyasaatAhmed** [reply]
Okay, thanks @amyegan I'll create a new discussion there.
